pkg/distributor/distributor.go

```diff
@@ -1208,6 +1208,9 @@ func (d *Distributor) prePushValidationMiddleware(next PushFunc) PushFunc {
 		var firstPartialErr error
 		var removeIndexes []int
 		totalSamples, totalExemplars := 0, 0
+		// Cap how many metrics with dedupled samples to emit trace events for.
+		const maxTrace = 10
+		dedupedPerMetric := make(map[string]int, maxTrace)
```

Comment:
Can we avoid making this allocation in the happy path?

Reply:
Done.